### ISSUE TITLE
Pose Solver in Unity

### ISSUE THREAD

**Issue by FarzanAlipour** [vignetteapp/Moetion#17]
Hey guys,
I've been trying to implement Moetion logic on Unity but the logic doesn't seem to work.
I think you are using different orientation and different 3d rigs.
Unity uses left handed (Y axis up) and I'm using mixamo rigs,
please inform me about the used orientation and rig so I can translate the code for Unity.
thanks ^^

**Comment by sr229** [comment]
Please see if you have the same issue as GH-11. We have a different representation for angles.

**Comment by FarzanAlipour** [comment]
I confirm that I convert radians to degrees
can you briefly explain the representation for angles?

**Comment by sr229** [comment]
It's most likely the rig you're using, we have not encountered this problem during testing.

**Comment by sr229** [comment]
Marking wontfix as this seems to be more of a human-induced error contributed by its usage on a unsupported environment.

**Comment by FarzanAlipour** [comment]
Hey again
I didn't ask for a fix!!
I asked for information on the used rig (I'm using mixamo characters) and the axis orientation on this package.
it would be great if you can provide these informations. otherwise your answer was not helpful at all.
anyway thanks for responding.

**Comment by sangww** [comment]
Hi I am also bumping this up, as I have run into the same issue as @FarzanAlipour. Mixamo rig seems to not utilize the same coordinate system for rotation (more precisely, local rotation) and wonder if there could be tutorial on rigging based on the avatar model you guys have used.

**Comment by sr229** [comment]
I apologise but the fact you are using an early stage library we have never personally tested yet (since we're still working on our own project and as such we cannot test it), and we only put up the packages for our own use later, means you have not taken appropriate steps to debug and fix under your own power.

We may fix it ourselves only if we confirmed it on our side but for now please fix it yourself if you can.

**Comment by Speykious** [comment]
Just wanted to step in a bit, Moetion is supposed to be a 1:1 port of [Kalidokit](https://github.com/yeemachine/kalidokit/) from Typescript to C#. So it should work relatively the same way. I personally didn't pay attention to how the angles work exactly when I translated this to C# and haven't had the occasion to test it when I was still working on it, however you may find the answer if you look in Kalidokit.

**Comment by FarzanAlipour** [comment]
Ok thanks @sr229 @Speykious 
I will opened the same issue on Kalidokit, and will update here if i found a solution.

**Comment by Speykious** [comment]
> Ok thanks @sr229 @Speykious I will opened the same issue on Kalidokit, and will update here if i found a solution.

Just make sure to test whether your Mixamo rigs work with Kalidokit before opening an issue about it there.

**Comment by sangww** [comment]
Hi all thanks for confirming this! It is great to understand the scope of this development, and apology if I came across demanding. I just hoped to point out an issue I also came across when it comes to Moetion->Unity->Mixamo adaptation.

@FarzanAlipour Here, I think I found a way to work. Further test needed, but was able to apply to one important joint for me: upper arm.

I have looked into in detail what the local coordinate system of the upper shoulder of the mixamo rig is (e.g. mixamorig:RightShoulder) and found that the coordinate system is rotated away from the x, y, z definition from MediaPipe or Moetion. There I applied transformation by switching out x, y, z accordingly, based on on what global axis rotation is applied to the local rotation. For instance, that swapping is done through a code that looks like below:

      Quaternion x = Quaternion.AngleAxis(ea.z * Mathf.Rad2Deg, Vector3.right);
      Quaternion y = Quaternion.AngleAxis(ea.x * Mathf.Rad2Deg, Vector3.up);
      Quaternion z = Quaternion.AngleAxis(ea.y * Mathf.Rad2Deg, Vector3.forward);

      return z * y * x;

This is for the upper right arm, but others will follow the same structure. I am further testing today, but no deadline tbh. Hope this helps you!
